### PR TITLE
Fiori ngx for generic list

### PR DESCRIPTION
**Pull request by parostatkiem**
**Description**
I had to do a little hack (safe one 💂 ) to disable _click_ event propagation in `Popover` component.
I've already submitted [an issue](https://github.com/SAP/fundamental-ngx/issues/479) for that to Fundamental-ngx.
Changes proposed in this pull request:

- Introduce Fundamental-ngx into `list-element-actions`, `list-filter`, `list-search`, `paging`
- Fix e2e tests
- Fix `ListElementActionsComponent.entryAsString` function returning _[object Object]_ in some case

**Related issue(s)**

### CODE REVIEW

**Comment by parostatkiem** [comment]
@dariadomagala This is purely Fundamental-ngx issue. I can try to fix it but it's not in scope of this task, I believe.

File: components/angular/generic-list/package.json [hunk 1]
```diff
@@ -10,6 +10,9 @@
   },
   "private": true,
   "dependencies": {
+    "@angular/router": "^7.2.1",
```
[the remark this answers is not in the document]
Comment (parostatkiem) [reply components/angular/generic-list/package.json]: Yes, we do. (checked it)

File: lambda/src/styles.scss [hunk 2]
```diff
@@ -91,34 +88,45 @@ $sf-dark-color-red: #A92121;
...
     margin-top: 32px;
   }
 }
 
 h2.sf-section__headline {
-	margin-top: 32px;
+  margin-top: 32px;
+}
+
+.fd-input-group {
```
Comment (dariadomagala) [inline lambda/src/styles.scss]: Could we overwrite styles for input here:
screen shot 2019-02-05 at 12 02 03
It doesn't look nice, and I think this frame is not visible in other inputs.

Comment (parostatkiem) [reply lambda/src/styles.scss]: I agree the whole label-entering component needs to be redesigned using f-ngx but it's definitely not in scope of this task. This element is not affected by the code above :)

File: components/angular/generic-list/src/app/modules/list/list-element-actions/list-element-actions.component.html [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
-          aria-haspopup="true" aria-label="More" (click)="toggleActionsBtn($event)">
-    <span class="tn-icon tn-icon--more tn-icon--medium" role="presentation"></span>
-  </button>
-  <ul class="tn-dropdown__menu tn-contextual-menu text-left" [attr.aria-hidden]="ariaHidden" [id]="entryAsString()">
-    <li *ngFor="let action of actions" >
-      <a (click)="closeActionsBtn($event); executeAction(action, $event);" class="tn-dropdown__item" [name]="action.name">{{ action.name }}</a>
-    </li>
-  </ul>
-</div>
+<fd-popover [alignment]="'right'" (click)="handlePopoverClick($event);" #popover>
```
[the remark this answers is not in the document]
Comment (parostatkiem) [reply components/angular/generic-list/src/app/modules/list/list-element-actions/list-element-actions.component.html]: Great solution! Thanks for saving my time :)

File: components/angular/generic-list/src/app/modules/list/list-element-actions/list-element-actions.component.ts [hunk 4]
```diff
@@ -12,19 +12,21 @@ import {
   styleUrls: ['./list-element-actions.component.scss'],
 })
 export class ListElementActionsComponent {
+  // tslint:disable:no-unnecessary-semicolons
```
Comment (dariadomagala) [inline components/angular/generic-list/src/app/modules/list/list-element-actions/list-element-actions.component.ts]: Please remove it.

File: components/angular/generic-list/src/app/modules/list/list-element-actions/list-element-actions.component.ts [hunk 5]
```diff
@@ -12,19 +12,21 @@ import {
...
   @Input() entryEventHandler: any;
   @Input() actions: any[];
   @ViewChild('popover') private popover: any;
 
   constructor(private changeDetector: ChangeDetectorRef) {}
 
-  handlePopoverClick(event) {
+  handlePopoverClick(event: Event): void {
     event.stopPropagation();
+    this.fireClick(document);
```
Comment (dariadomagala) [inline components/angular/generic-list/src/app/modules/list/list-element-actions/list-element-actions.component.ts]: There are some errors that might be related to that. I would vote for removing that part and allow multiple dropdowns to be open (for now). This issue is blocking others so please create a separate ticket to fix that part, and we will merge it as it is currently.

File: components/angular/generic-list/src/app/modules/list/list-filter/list-filter.component.html [hunk 6]
```diff
@@ -1,16 +1,23 @@
-<div class="tn-dropdown" (clickOutside)="autoCloseDropdown()" clickOutsideEvents="click,mousedown">
-    <button class="tn-button tn-button--small tn-button--text"
-            [ngClass]="{'tn-button--disabled': !hasFacets()}"
-            (click)="toggleDropdown($event)"
-            [attr.aria-expanded]="ariaExpanded">
+<fd-popover [alignment]="'right'" *ngIf="filterState.availableFacets.length">
+  <fd-popover-control>
+    <button fd-button [fdType]="'toolbar'" [glyph]="'filter'" [ngClass]="{ 'tn-button--disabled': !hasFacets() }"
```
[the remark this answers is not in the document]
Comment (parostatkiem) [reply components/angular/generic-list/src/app/modules/list/list-filter/list-filter.component.html]: I'll create one.

File: components/angular/generic-list/src/app/modules/list/list-element-actions/list-element-actions.component.ts (hunk 5 above)
Comment (parostatkiem) [reply components/angular/generic-list/src/app/modules/list/list-element-actions/list-element-actions.component.ts]: Couldn't reproduce :(

File: components/angular/generic-list/README.md [hunk 7]
```diff
@@ -11,25 +11,37 @@ A sandbox application appears. It uses `<y-generic-table>` and `<y-generic-list>
...
 To use the package in a component, perform the following steps:
 
-1. Add `"preserveSymlinks": true` to `angular.json` test and build options. Or alternatively add `--preserve-symlink` option to `ng test` and `ng build`. 
+1. Run 
+    ```bash
+    npm install --save fiori-fundamentals fundamental-ngx
+    ```
+   in your project directory.
+
+2. Add `"preserveSymlinks": true` to `angular.json` test and build options. Or alternatively add `--preserve-symlink` option to `ng test` and `ng build`.
```
Comment (bszwarc) [inline components/angular/generic-list/README.md]:
```suggestion
2. Add `"preserveSymlinks": true` to `angular.json` test and build options. Alternatively, add `--preserve-symlink` option to `ng test` and `ng build`. 
```

File: components/angular/generic-list/src/app/modules/list/list-element-actions/list-element-actions.component.ts [hunk 8]
```diff
@@ -21,16 +20,22 @@ export class ListElementActionsComponent {
   constructor(private changeDetector: ChangeDetectorRef) {}
 
   handlePopoverClick(event: Event): void {
+    const wasPopoverOpened = this.popover && this.popover.isOpen;
```
Comment (dariadomagala) [inline components/angular/generic-list/src/app/modules/list/list-element-actions/list-element-actions.component.ts]: maybe `isPopoverOpened`? :)

Comment (parostatkiem) [reply components/angular/generic-list/src/app/modules/list/list-element-actions/list-element-actions.component.ts]: No, it is changed many times during the function execution and only the initial state matters.

**Pull request merged**